### ISSUE TITLE
Data checker fails on manifest of org.freedesktop.Sdk.Extension.texlive

### ISSUE THREAD

**Issue by jakobjakobson13** [flathub-infra/flatpak-external-data-checker#88]
Hi,

the `flatpak-external-data-checker` fails on the manifest of the [https://github.com/flathub/org.freedesktop.Sdk.Extension.texlive/blob/branch/19.08/org.freedesktop.Sdk.Extension.texlive.yml](https://github.com/flathub/org.freedesktop.Sdk.Extension.texlive/blob/branch/19.08/org.freedesktop.Sdk.Extension.texlive.yml) as the newer version of perl does not get reported.
In addition, the local flatpak version [https://github.com/flathub/org.flathub.flatpak-external-data-checker](https://github.com/flathub/org.flathub.flatpak-external-data-checker) just ends after the texlive part with the following output:

```
+ 2020-06-21 20:18:48,116   DEBUG src.checker: [3/6] checking texlive2020.iso
+ 2020-06-21 20:18:48,116   DEBUG src.checkers.debianrepochecker: texlive2020.iso is not a debian-repo type ext data
+ 2020-06-21 20:18:48,116   DEBUG src.checkers.flashchecker: texlive2020.iso is not a flash type ext data
+ 2020-06-21 20:18:48,116   DEBUG src.checkers.htmlchecker: Getting extra data info from http://mirror.ctan.org/systems/texlive/Images; may take a while
+ 2020-06-21 20:18:48,492   DEBUG src.checkers.htmlchecker: version-pattern re.compile('texlive(\\d\\d\\d\\d).iso') matched multiple times, selecting latest
+ 2020-06-21 20:18:48,493   DEBUG src.checkers.htmlchecker: version-pattern re.compile('texlive(\\d\\d\\d\\d).iso') matched: 2020
$
```

Is this due to the size of the iso file or to its file type?

**Comment by wjt** [comment]
> the flatpak-external-data-checker fails on the manifest of the https://github.com/flathub/org.freedesktop.Sdk.Extension.texlive/blob/branch/19.08/org.freedesktop.Sdk.Extension.texlive.yml as the newer version of perl does not get reported.

I had a quick look and the following works for Perl:

```yaml
        x-checker-data:
          type: html
          url: https://www.cpan.org/src
          version-pattern: https://www.cpan.org/src/5.0/perl-(\d+\.\d+\.\d+).tar.gz
          url-template: https://www.cpan.org/src/5.0/perl-$version.tar.gz
```

From https://github.com/flathub/org.freedesktop.Sdk.Extension.texlive/commit/c5e7d15b2dcad924632418672d5ac6ce67475a70 this generates this update:

```yaml
+        url: https://www.cpan.org/src/5.0/perl-5.32.0.tar.gz
+        sha256: efeb1ce1f10824190ad1cadbcccf6fdb8a5d37007d0100d2d9ae5f2b5900c0b4
+        size: 18204616
```

It appears you've discovered the same.

>  In addition, the local flatpak version https://github.com/flathub/org.flathub.flatpak-external-data-checker just ends after the texlive part with the following output:
> Is this due to the size of the iso file or to its file type?

Hard to know. Check the exit status (`$?` in the shell).

**Comment by jakobjakobson13** [comment]
> 
> > In addition, the local flatpak version https://github.com/flathub/org.flathub.flatpak-external-data-checker just ends after the texlive part with the following output:
> > Is this due to the size of the iso file or to its file type?
> 
> Hard to know. Check the exit status (`$?` in the shell).

There is no exit status printed on the shell, neither with the normal command nor with the `-v` option.

**Comment by wjt** [comment]
After the checker exits abruptly, what does:

```
echo $?
```

show?

**Comment by jakobjakobson13** [comment]
The exit code is 137.
```
$ echo $?
137
$
```

and systemd logs the following lines that (at least I think so) are perhaps related to it:
```
kernel: Out of memory: Killed process 10617 (python3) total-vm:7869088kB, anon-rss:4490132kB, file-rss:0kB, shmem-rss:0kB, UID:1000 pgtables:10952kB oom_score_adj:0
kernel: oom_reaper: reaped process 10617 (python3), now anon-rss:0kB, file-rss:0kB, shmem-rss:0kB
```

Btw: The texlive iso is 3.7 GB big.

**Comment by wjt** [comment]
I think @gasinvein fixed this in #153 – the file size is likely the problem. It now avoids storing enormous files in memory. Please reopen if the problem persists.
